dfdatetime/posix_time.py

```diff
@@ -109,6 +111,106 @@ def CopyToDateTimeString(self):
         year, month, day_of_month, hours, minutes, seconds)
 
 
+class PosixTimeInMilliseconds(interface.DateTimeValues):
+  """POSIX timestamp in milliseconds.
+
+  Variant of the POSIX timestamp in milliseconds.
+
+  Attributes:
+    is_local_time (bool): True if the date and time value is in local time.
+  """
+
+  _EPOCH = PosixTimeEpoch()
+
+  def __init__(self, timestamp=None):
+    """Initializes a POSIX timestamp in milliseconds.
+
+    Args:
+      timestamp (Optional[int]): POSIX timestamp in milliseconds.
+    """
+    super(PosixTimeInMilliseconds, self).__init__()
+    self._precision = definitions.PRECISION_1_MILLISECOND
+    self._timestamp = timestamp
+
+  @property
+  def timestamp(self):
+    """int: POSIX timestamp in milliseconds or None if timestamp is not set."""
+    return self._timestamp
+
+  def _GetNormalizedTimestamp(self):
+    """Retrieves the normalized timestamp.
+
+    Returns:
+      decimal.Decimal: normalized timestamp, which contains the number of
+          seconds since January 1, 1970 00:00:00 and a fraction of second used
+          for increased precision, or None if the normalized timestamp cannot be
+          determined.
+    """
+    if self._normalized_timestamp is None:
+      if (self._timestamp is not None and self._timestamp >= self._INT64_MIN and
```

Comment:
Why add bounds checks here? There's nothing essential to "milliseconds from posix epoch" that limits the value to a 64 bit int. It makes sense to have this for specific implementations though, like Java time